src/com/xilinx/rapidwright/util/function/ThrowingConsumer.java

```diff
@@ -0,0 +1,30 @@
+/* 
+ * Copyright (c) 2022 Xilinx, Inc. 
+ * All rights reserved.
+ *
+ * Author: Jakob Wenzel, Xilinx Research Labs.
+ *  
+ * This file is part of RapidWright. 
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * 
+ */
+ 
+package com.xilinx.rapidwright.util.function;
+/**
+ * Like {@link java.util.function.Consumer}, but with checked exceptions
+ */
+@FunctionalInterface
+public interface ThrowingConsumer <T, E extends Throwable> {
```

Comment:
Doesn't look like this interface is being used. Is it present for completeness?

Reply:
Correct, I added them for completeness.

However, I am now aware of the interfaces in `org.apache.commons.io.function`. I'm essentially duplicating them, so I might as well switch to those.